t/distribution-depends-parsing.t

```diff
@@ -49,17 +49,66 @@ with Zef::Distribution.new(|Rakudo::Internals::JSON.from-json(q:to/META6/)) -> $
         "description":"Test hash-based depends and native depends parsing",
         "license":"none",
         "depends": {
-            "by-distro.name": {
-                "win32": [ "Foo::Win32" ],
-                "linux": [ "Foo::Linux" ],
-                "" : [ "Foo::Unknown" ]
+            "runtime" : {
```

Comment:
@niner note how the previous test did not have the required `require` section of the hash form of depends. I don't know how this test used to pass, but I suspect that whatever did was what allowed the `[ "Foo::Win32" ]` instead of the `"Foo::Win32"` I had to use to get this to work.

Reply:
Ah, that's easy: it's wasn't a Hash style depends at all. The whole Hash got replaced by one of the arrays containing the single module name, so it ended up as `"depends": [ "Foo::Linux" ]`
Easy to explain, but it did take me quite a while to notice...

So the test was actually correct as it was explicitly designed to do this according to the commit message:
"Allow for a distro's whole dependency list to be system specific        
Apply system-collapse even to the top level depends value."

Reply:
> So the test was actually correct as it was explicitly designed to do this according to the commit message:
"Allow for a distro's whole dependency list to be system specific
Apply system-collapse even to the top level depends value."

The test was correct in that you can have multiple depends in that (quasi) leaf node, but it was coincidental that not having the `requires` section worked.

Reply:
Actually I think I see how the original message could infer the original behavior -- if there was an e.g. `optional` field in addition to the `requires` then you could have all of them inside the conditional. However, it does seem coincidental that the `requires` isn't then needed inside the conditional.